src/pages/AdminPages/AdminDashboard/index.jsx

```diff
@@ -1,32 +1,63 @@
-import { useHistory } from 'react-router-dom';
-
+import { useDispatch, shallowEqual } from 'react-redux';
+import { 
+    getTraders, getUsers, getAdmins,
+    getAffiliates, setAlterUsers, getUnapprovedUsers
+} from "../../../store/reducers/adminQuery";
+import AlterUsers from '../../../components/Admin/AlterUsers';
+import { setIsLoading } from '../../../store/reducers/user';
 
 const AdminDashboard = () => {
-    const history = useHistory()
-    const { role, isLoading } = useSelector(state => ({
+    const [removeEntry, setRemoveEntry] = useState(false)
+    const dispatch = useDispatch()
+    const { role, isLoading, alterUsers, traders } = useSelector(state => ({
         role : state.user.role,
-        isLoading : state.isLoading
-    }))
+        isLoading : state.user.isLoading,
+        alterUsers: state.adminQuery.alterUsers,
+        traders: state.adminQuery.traders,
+    }), shallowEqual)
+
+    useEffect(() => {
+        dispatch(setIsLoading());
+        dispatch(getUsers())
+        dispatch(getTraders())
+        // eslint-disable-next-line
+    }, [])
+
+    const update = (remove, dispatchFunc) => {
+        dispatch(setIsLoading());
+        setRemoveEntry(remove)
+        dispatch(dispatchFunc())
+    }
         
 
     return(
         <>
             <Authenticate inside={true} />
 
             {
-                !isLoading && role && (
+                role && (
                 
                 role.isAdmin ? (
                     <>
 
                         This is the Admin Dashboard Page
                         <hr />
-                        <button onClick={() => history.push("/admin/view/users/unapproved")}>View Unapproved Accounts</button>
+                        <button disabled={alterUsers[0] && !alterUsers[0].isApproved && true} onClick={() => {update(true, getUnapprovedUsers)}}>View Unapproved Accounts</button> <br />
+                        <button disabled={alterUsers[0] && alterUsers[0].role.isUser && !alterUsers[0].role.isAffiliate  && false} onClick={() => {update(false, getUsers)}}>View User Accounts</button> <br />
+                        <button disabled={alterUsers[0] && alterUsers[0].role.isAffiliate  && true} onClick={() => {update(false, getAffiliates)}}>View Affiliate Accounts</button> <br />
+                        <button disabled={alterUsers[0] && alterUsers[0].role.isTrader  && true} onClick={() => {setRemoveEntry(false); dispatch(setAlterUsers(traders))}}>View Traders Accounts</button> <br />
+                        <button disabled={alterUsers[0] && alterUsers[0].role.isAdmin  && true} onClick={() => {update(false, getAdmins);}}>View Admin Accounts</button> <br />
 
+                        <hr />
+                        {
+                        isLoading ? 
+                        <>Loading...</>:
+                            alterUsers.length > 0 ? <AlterUsers users={alterUsers} approve={removeEntry} /> : <>No User Found For This Category</>
+                        }
```

Comment:
If i were to guess, you are now loading the table of users on the dashboard Page... that's a big no!

We can't deteremine what will be on dashboard now.... It will be preferable to have a separate page to show all the different kind of users.

If we are then going with the route of having all users showcase on one page, there'll be no use of putting AlterUsers and SelectedUserModal components in the Component folder since they'll only be used by this one page.

Reply:
Well, we can have a couple of data on the dashboard page. So having this data on the dashboard page isn't bad, concerning the file location I can move the folder from the `components` folder and place inside the `Admin` components in `pages`.

Reply:
At this point, yes, you are right... but lay no focus on the dashboard, we can't know what should be there until we are done with the app.

Recreate a 2 new pages... one for approving accounts and another for edit accounts... there's a whole lot more that can be viewed in the edit user page...it's literally going to be an information page for view user's activity logs, transaction logs, active portfolios etc....

Reply:
Alright, thanks.
Can you suggest a nice URL for these two?
I'm thinking instead of haaving 2 files for these 2 things, I'd just use 1 and determine what the admin wants to request for from the URL then use that to fetch the right data